### PR TITLE
added feature of autogrowing when editor gets focus

### PR DESCRIPTION
**Pull request by DegJ**
Added a feature of allowing the editor to shrink when the editor blurs and grow again when it gets focus.

### CODE REVIEW

File: src/trumbowyg.js [hunk 1]
```diff
@@ -1032,6 +1051,15 @@ jQuery.trumbowyg = {
                     t.$c.trigger('tbwresize');
                 }
             }
+            if (t.o.autogrowOnEnter) {
+                t.$ed.height('auto');
+                var totalheight = t.autogrowOnEnter_hasHadFocus ? t.$ed[0].scrollHeight : 0;
+                if (totalheight !== t.$ta.css('height')) {
+                    t.$ta.css({ height: totalheight });
+                    t.$ed.css({ height: totalheight });
```
Comment (Alex-D) [inline src/trumbowyg.js]: Use `t.$ta.add(t.$ed).css...` to avoid code duplication :)

File: src/trumbowyg.js [hunk 2]
```diff
@@ -608,6 +609,24 @@ jQuery.trumbowyg = {
                     if (e.type === 'blur') {
                         $('.' + prefix + 'active-button', t.$btnPane).removeClass(prefix + 'active-button ' + prefix + 'active');
                     }
+                    if (t.o.autogrowOnEnter) {
+                        if (e.type === 'focus') {
+                            t.autogrowOnEnter_hasHadFocus = true;
```
Comment (Alex-D) [inline src/trumbowyg.js]: This name does not fit other var names of the project :/

You can just name that `autogrowOnEnterWasFocused`

**Comment by Alex-D** [comment]
By the way, that's a good addition :)

**Comment by DegJ** [comment]
@Alex-D I'll fix the issues. Should have waited with the PR a few more :)

**Comment by DegJ** [comment]
@Alex-D should be working alot better now and conform alot more. (might have done a editor wide format, thus fixing a few spacing issues throughout the code (didnt see it until i was done and already had commited it and made alot of changes)

**Comment by Alex-D** [comment]
Thanks! :)

**Pull request merged**